libcentrifugo/api.go

```diff
@@ -100,19 +101,29 @@ func (app *Application) broadcastCmd(cmd *broadcastAPICommand) (*response, error
 	resp := newResponse("broadcast")
 	channels := cmd.Channels
 	data := cmd.Data
-	var err error
 	if len(channels) == 0 {
 		logger.ERROR.Println("channels required for broadcast")
 		resp.Err(ErrInvalidMessage)
 		return resp, nil
 	}
+	var wg sync.WaitGroup
+	var firstErr error
 	for _, channel := range channels {
-		err = app.publish(channel, data, cmd.Client, nil, false)
-		if err != nil {
-			logger.ERROR.Println("Error publishing into channel", string(channel))
-			resp.Err(err)
-			return resp, nil
-		}
+		wg.Add(1)
+		go func(channel Channel) {
```

Comment:
Hmm, this works but it feels off to me...

It's intentionally creating overhead and contention on the publish channel etc as well as spawning maybe thousands of goroutines. Perhaps that's not too bad in practice, but in theory it would be possible to do this much more efficiently if we had a real async API in our engine interface.

Maybe something for the future?